### PR TITLE
LPS-197067 Add the roles when inherited and do not filter them in the SearchPermissionChecker

### PR DESCRIPTION
**Pull request by robertoDiaz**
LPS-197067 Add the roles when inherited and do not filter them in the SearchPermissionChecker

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

File: modules/apps/depot/depot-service/src/main/java/com/liferay/depot/internal/security/permission/contributor/DepotRoleContributor.java [hunk 1]
```diff
@@ -85,6 +93,31 @@ private void _addRoleId(RoleCollection roleCollection, String roleName)
...
+			_userGroupLocalService.getGroupUserGroups(groupId);
+
+		List<UserGroup> userUserGroups =
+			_userGroupLocalService.getUserUserGroups(userBag.getUserId());
+
+		Stream<UserGroup> groupUserGroupsStream = groupUserGroups.stream();
+
+		if (groupUserGroupsStream.anyMatch(userUserGroups::contains)) {
+			return true;
+		}
```
Comment (adolfopa) [inline modules/apps/depot/depot-service/src/main/java/com/liferay/depot/internal/security/permission/contributor/DepotRoleContributor.java:116]: Stream usage is discouraged. Can you replace this with a regular for loop?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/depot/depot-service/src/main/java/com/liferay/depot/internal/security/permission/contributor/DepotRoleContributor.java (hunk 1 above)
Comment (robertoDiaz) [reply modules/apps/depot/depot-service/src/main/java/com/liferay/depot/internal/security/permission/contributor/DepotRoleContributor.java:116]: ok :(

**Comment by austinchiang** [comment]
Yeah doesn't look related

**Comment by austinchiang** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by robertoDiaz** [comment]
@liferay-lima 

could anyone push this to Brian?

Thanks!

**Comment by adolfopa** [comment]
ci:forward:force

**Pull request closed without merging**